### PR TITLE
Nitpicking Patrick's Lab

### PR DESCRIPTION
**Pull request by cwallac**
Since you finished this up in short order, if you wish to, I'll find some likely small things you can improve on.

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cwallac) [inline .idea/compiler.xml:1]: We will want to modify the gitignore to ignore this directory

File: app/app.iml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cwallac) [inline app/app.iml:1]: Same with this file no .iml extensions

File: app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile 'com.android.support:appcompat-v7:22.2.1'
+    compile 'com.github.alexkolpa:floating-action-button-toolbar:0.5.1'
+    compile 'com.melnykov:floatingactionbutton:1.3.0'
+    compile 'com.github.ozodrukh:CircularReveal:1.1.1@aar'
+    compile 'com.jakewharton:butterknife:7.0.1'
```
Comment (cwallac) [inline app/build.gradle:28]: Well you beat us to introducing this library to the class.

File: app/src/main/java/com/android/phuston/noted/ArrayHelper.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    public void saveArray(String key, ArrayList<String> array) {
+        JSONArray jArray = new JSONArray(array);
+        editor.remove(key);
+        editor.putString(key, jArray.toString());
+        editor.commit();
+    }
+
+    public ArrayList<String> getArray(String key) {
+        ArrayList<String> array = new ArrayList<String>();
+        String jsonArrayString = prefs.getString(key, "NO_PREF_SAVED");
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/ArrayHelper.java:35]: make this a private static final String and define it at the top of the class

File: app/src/main/java/com/android/phuston/noted/ArrayHelper.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                return array;
+            } catch (JSONException e) {
+                return getDefaultArray();
+            }
+        }
+    }
+
+    public ArrayList<String> getDefaultArray(){
+        ArrayList<String> array = new ArrayList<String>();
+        array.add("Add a new note!");
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/ArrayHelper.java:53]: If you are displaying these to the user you will want to put them in the strings.xml file so they can be easily translated

File: app/src/main/java/com/android/phuston/noted/NoteFragment.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+        listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                final int pos = position;
+
+                AlertDialog.Builder alertBuilder = new AlertDialog.Builder(getActivity());
+
+                alertBuilder.setTitle("Edit Note");
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/NoteFragment.java:55]: Put this in String.xml

File: app/src/main/java/com/android/phuston/noted/NoteFragment.java [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                alertBuilder.setMessage("Type your edited note below.");
+
+                //Set EditText view to get user input
+                final EditText mInput = new EditText(getActivity());
+                mInput.setText((String) listview.getItemAtPosition(pos));
+                alertBuilder.setView(mInput);
+
+                alertBuilder.setPositiveButton("Add", new DialogInterface.OnClickListener() {
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/NoteFragment.java:63]: String.xml

Also define this OnClickListener as a separate class that implements OnClickListener so your fragment stays clean of these lines of code.

File: app/src/main/java/com/android/phuston/noted/NoteFragment.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        mNotes.set(pos, mInput.getText().toString());
+                    }
+                });
+
+                alertBuilder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int whichButton) {
+                        // Canceled.
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/NoteFragment.java:72]: Does this actually close the alert dialog? I had always thought you may have to call dismiss()

File: app/src/main/java/com/android/phuston/noted/NoteFragment.java [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+                alertToShow.show();
+            }
+        });
+
+        listview.setAdapter(mNotesAdapter);
+
+        fab.attachToListView(listview);
+
+        fab.setOnClickListener(new View.OnClickListener() {
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/NoteFragment.java:89]: Again, if you define this in a separate class itll make your fragment much cleaner

File: app/src/main/java/com/android/phuston/noted/NoteFragment.java [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+            @Override
+            public void onClick(View v) {
+                AlertDialog.Builder alertBuilder = new AlertDialog.Builder(getActivity());
+
+                alertBuilder.setTitle("New Note");
```
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/NoteFragment.java:94]: Put it in Strings.xml

File: app/src/main/java/com/android/phuston/noted/NoteFragment.java (hunk 6 above)
Comment (sihrc) [inline app/src/main/java/com/android/phuston/noted/NoteFragment.java:55]: option + enter on string literals in android studio will give you the option to quickly extract to a string resource

**Comment by sihrc** [comment]
quick rebase will make your commit history beautiful

File: app/src/main/java/com/android/phuston/noted/ArrayHelper.java (hunk 4 above)
Comment (cwallac) [inline app/src/main/java/com/android/phuston/noted/ArrayHelper.java:35]: By this I mean "NO_PREF_SAVED"

**Pull request closed without merging**